clever_tools/src/flight_routines/__init__.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+import sys
+import csv
+import math
+import time
+import logging
+import threading
+import rospy
+import copy
+from clever import srv
+from mavros_msgs.srv import SetMode
+from mavros_msgs.srv import CommandBool
+from std_srvs.srv import Trigger
+
+# Create proxies to services
+navigate = rospy.ServiceProxy('/navigate', srv.Navigate)
+navigate_global = rospy.ServiceProxy('/navigate_global', srv.NavigateGlobal)
+set_position = rospy.ServiceProxy('/set_position', srv.SetPosition)
+set_velocity = rospy.ServiceProxy('/set_velocity', srv.SetVelocity)
+set_attitude = rospy.ServiceProxy('/set_attitude', srv.SetAttitude)
+set_rates = rospy.ServiceProxy('/set_rates', srv.SetRates)
+get_telemetry = rospy.ServiceProxy('/get_telemetry', srv.GetTelemetry)
+land = rospy.ServiceProxy('/land', Trigger)
+arming = rospy.ServiceProxy('/mavros/cmd/arming', CommandBool)
+set_mode = rospy.ServiceProxy('/mavros/set_mode', SetMode)
+
+rospy.loginfo("Proxy services inited")
+
+# Globals
+FREQUENCY = 10              # hz
+TOLERANCE = 0.2             # m
+SPEED = 1.0                 # m/s
+TAKEOFF_SPEED = 1.0         # m/s
+TAKEOFF_HEIGHT = 1.0        # m
+LOCAL_FRAME_ID = 'map'
+COPTER_FRAME_ID = 'body'
+
+# Get distance betwen 2 points
+def get_distance(x1, y1, z1, x2, y2, z2):
+    return math.sqrt((x1 - x2) ** 2 + (y1 - y2) ** 2 + (z1 - z2) ** 2)
+
+# Takeoff to specified height
+def takeoff(height=TAKEOFF_HEIGHT, speed=TAKEOFF_SPEED, tolerance=TOLERANCE, frame_id=LOCAL_FRAME_ID):
+    rospy.loginfo("Takeoff started...")
+    rate = rospy.Rate(FREQUENCY)
+    start = get_telemetry(frame_id=frame_id)
+    climb = 0.
+    result = navigate(x=start.x, y=start.y, z=start.z+height, speed=speed, yaw=float('nan'), frame_id=frame_id, auto_arm=True)
+    rospy.loginfo(result)
+    while abs(climb - height) > tolerance:
+        climb = abs(get_telemetry(frame_id=frame_id).z - start.z)
+        rospy.loginfo("Takeoff to {:.2f} of {:.2f} meters".format(climb, height))
+        rate.sleep()
+    rospy.loginfo("Takeoff succeeded!")
+
+# Reach specified point. Copter needs to be armed.
+def reach_point(x, y, z, yaw=float('nan'), speed=SPEED, tolerance=TOLERANCE, frame_id=LOCAL_FRAME_ID):
```

Comment:
Actually, we have a more universal version of this function, utilizing `navigate_target` frame: see https://clever.copterexpress.com/ru/snippets.html#block-nav.